urllib3/response.py

```diff
@@ -394,20 +401,23 @@ def getheader(self, name, default=None):
     def close(self):
         if not self.closed:
             self._fp.close()
+            self._fp = None
 
         if self._connection:
             self._connection.close()
 
     @property
     def closed(self):
+        # TODO: Do we need this?
         if self._fp is None:
             return True
-        elif hasattr(self._fp, 'closed'):
-            return self._fp.closed
+        elif hasattr(self._fp, 'complete'):
+            return self._fp.complete
         else:
-            return True
+            return False
 
     def fileno(self):
+        # TODO: Do we need this?
```

Comment:
> I don't think we want users to select on our response object, do we?

No I don't think we do. Especially not since we're selecting on the underlying parts of this.